plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -201,6 +205,9 @@
 
     private final Map<String, Optional<Long>> snapshotIds = new ConcurrentHashMap<>();
     private final Map<SchemaTableName, TableMetadata> tableMetadataCache = new ConcurrentHashMap<>();
+
+    private final LoadingCache<String, ConcurrentHashMap<SchemaTableName, Table>> tableCache;
```

Comment:
Nested keying `query id -> SchemaTableName -> Table` is redundant.
Make `SchemaTableName` the key here, as in `tableMetadataCache` (and effectively in `snapshotIds`, although the code doesn't make it obvious)